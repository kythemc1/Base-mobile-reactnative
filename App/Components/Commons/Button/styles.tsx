import {StyleSheet} from 'react-native';
import sizes from 'Configs/UI/Sizes';
import {colors} from 'Configs/UI/Colors';

export const styles = StyleSheet.create({
    mContainerSecondaryCustom: {
        borderRadius: sizes._12sdp,
        borderColor: colors.color_primary1,
        borderWidth: sizes._1sdp,
        backgroundColor: colors.color_white,
        alignItems: 'center',
        justifyContent: 'center'
    },
    mTextSecondaryCustom: {
        fontSize: sizes._16sdp,
        fontWeight: '600',
        lineHeight: sizes._24sdp,
        color: colors.color_primary1
    },
    mContainerPrimaryCustom: {
        borderRadius: sizes._12sdp,
        backgroundColor: colors.color_primary2,
        alignItems: 'center',
        justifyContent: 'center'
    },
    mTextPrimaryCustom: {
        fontSize: sizes._16sdp,
        fontWeight: '600',
        lineHeight: sizes._24sdp,
        color: colors.color_white
    },
    mContainerSecondaryLarge: {
        width: sizes._screenWidth * 0.9,
        height: sizes._screenHeight * 0.06,
        borderRadius: sizes._12sdp,
        borderColor: colors.color_primary1,
        borderWidth: sizes._1sdp,
        backgroundColor: colors.color_white,
        alignItems: 'center',
        justifyContent: 'center'
    },
    mTextSecondaryLarge: {
        fontSize: sizes._16sdp,
        fontWeight: '600',
        lineHeight: sizes._24sdp,
        color: colors.color_primary1
    },
    mContainerPrimaryLarge: {
        width: sizes._screenWidth * 0.9,
        height: sizes._screenHeight * 0.06,
        borderRadius: sizes._12sdp,
        backgroundColor: colors.color_primary2,
        alignItems: 'center',
        justifyContent: 'center'
    },
    mTextPrimaryLarge: {
        fontSize: sizes._16sdp,
        fontWeight: '600',
        lineHeight: sizes._24sdp,
        color: colors.color_white
    },
    mContainerSecondaryNormal: {
        width: sizes._screenWidth * 0.44,
        height: sizes._screenHeight * 0.06,
        borderRadius: sizes._12sdp,
        borderColor: colors.color_primary1,
        borderWidth: sizes._1sdp,
        backgroundColor: colors.color_white,
        alignItems: 'center',
        justifyContent: 'center'
    },
    mTextSecondaryNormal: {
        fontSize: sizes._16sdp,
        fontWeight: '600',
        lineHeight: sizes._24sdp,
        color: colors.color_primary1
    },
    mContainerPrimaryNormal: {
        width: sizes._screenWidth * 0.44,
        height: sizes._screenHeight * 0.06,
        borderRadius: sizes._12sdp,
        backgroundColor: colors.color_primary2,
        alignItems: 'center',
        justifyContent: 'center'
    },
    mTextPrimaryNormal: {
        fontSize: sizes._16sdp,
        fontWeight: '600',
        lineHeight: sizes._24sdp,
        color: colors.color_white
    },
    mContainerSecondarySmall: {
        width: sizes._screenWidth * 0.4,
        height: sizes._screenHeight * 0.05,
        borderRadius: sizes._12sdp,
        borderColor: colors.color_primary1,
        borderWidth: sizes._1sdp,
        backgroundColor: colors.color_white,
        alignItems: 'center',
        justifyContent: 'center'
    },
    mTextSecondarySmall: {
        fontSize: sizes._14sdp,
        fontWeight: '600',
        lineHeight: sizes._20sdp,
        color: colors.color_primary1
    },
    mContainerPrimarySmall: {
        width: sizes._screenWidth * 0.4,
        height: sizes._screenHeight * 0.05,
        borderRadius: sizes._12sdp,
        backgroundColor: colors.color_primary2,
        alignItems: 'center',
        justifyContent: 'center'
    },
    mTextPrimarySmall: {
        fontSize: sizes._14sdp,
        fontWeight: '600',
        lineHeight: sizes._20sdp,
        color: colors.color_white
    },
    mContainerSecondaryCompact: {
        width: sizes._screenWidth * 0.3,
        height: sizes._screenHeight * 0.03,
        borderRadius: sizes._12sdp,
        borderColor: colors.color_primary1,
        borderWidth: sizes._1sdp,
        backgroundColor: colors.color_white,
        alignItems: 'center',
        justifyContent: 'center'
    },
    mContainerNoLine: {
        width: sizes._screenWidth * 0.44,
        height: sizes._screenHeight * 0.06,
        borderRadius: sizes._12sdp,
        backgroundColor: colors.color_white,
        alignItems: 'center',
        justifyContent: 'center'
    },
    mTextNoLine: {
        fontSize: sizes._16sdp,
        fontWeight: '600',
        lineHeight: sizes._24sdp,
        color: colors.color_primary1
    },
    mTextSecondaryCompact: {
        fontSize: sizes._12sdp,
        fontWeight: '400',
        lineHeight: sizes._16sdp,
        color: colors.color_primary1
    },
    mContainerPrimaryCompact: {
        width: sizes._screenWidth * 0.3,
        height: sizes._screenHeight * 0.03,
        borderRadius: sizes._12sdp,
        backgroundColor: colors.color_primary2,
        alignItems: 'center',
        justifyContent: 'center'
    },
    mTextPrimaryCompact: {
        fontSize: sizes._12sdp,
        fontWeight: '400',
        lineHeight: sizes._16sdp,
        color: colors.color_white
    }
});
