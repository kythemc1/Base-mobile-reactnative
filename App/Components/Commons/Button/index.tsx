import {Text, TouchableOpacity} from 'react-native';
import {styles} from 'Components/Commons/Button/styles';
import sizes from 'Configs/UI/Sizes';

interface buttonInterface {
    text: string,
    typeButton: 'secondary' | 'primary',
    sizeButton: 'large' | 'normal' | 'small' | 'compact' | 'no_line' | 'custom',
    onPressButton: () => void
}

export const Button = (props: buttonInterface) => {

    return (
        <>
            {props.sizeButton == 'custom' ?
                props.typeButton == 'secondary' ?
                    <TouchableOpacity style={[styles.mContainerSecondaryCustom, {
                        marginHorizontal: sizes._16sdp,
                        width: sizes._screenWidth * 0.8,
                        height: sizes._56sdp
                    }]} onPress={props.onPressButton}>
                        <Text style={styles.mTextSecondaryCustom}>
                            {props.text}
                        </Text>
                    </TouchableOpacity> :
                    <TouchableOpacity style={[styles.mContainerPrimaryCustom, {
                        marginHorizontal: sizes._16sdp,
                        width: sizes._screenWidth * 0.8,
                        height: sizes._56sdp
                    }]} onPress={props.onPressButton}>
                        <Text style={styles.mTextPrimaryCustom}>
                            {props.text}
                        </Text>
                    </TouchableOpacity> : null}
            {props.sizeButton == 'large' ?
                props.typeButton == 'secondary' ?
                    <TouchableOpacity style={styles.mContainerSecondaryLarge} onPress={props.onPressButton}>
                        <Text style={styles.mTextSecondaryLarge}>
                            {props.text}
                        </Text>
                    </TouchableOpacity> :
                    <TouchableOpacity style={styles.mContainerPrimaryLarge} onPress={props.onPressButton}>
                        <Text style={styles.mTextPrimaryLarge}>
                            {props.text}
                        </Text>
                    </TouchableOpacity> : null}
            {props.sizeButton == 'normal' ?
                props.typeButton == 'secondary' ?
                    <TouchableOpacity style={styles.mContainerSecondaryNormal} onPress={props.onPressButton}>
                        <Text style={styles.mTextSecondaryNormal}>
                            {props.text}
                        </Text>
                    </TouchableOpacity> :
                    <TouchableOpacity style={styles.mContainerPrimaryNormal} onPress={props.onPressButton}>
                        <Text style={styles.mTextPrimaryNormal}>
                            {props.text}
                        </Text>
                    </TouchableOpacity> : null}
            {
                props.sizeButton == 'small' ?
                    props.typeButton == 'secondary' ?
                        <TouchableOpacity style={styles.mContainerSecondarySmall} onPress={props.onPressButton}>
                            <Text style={styles.mTextSecondarySmall}>
                                {props.text}
                            </Text>
                        </TouchableOpacity> :
                        <TouchableOpacity style={styles.mContainerPrimarySmall} onPress={props.onPressButton}>
                            <Text style={styles.mTextPrimarySmall}>
                                {props.text}
                            </Text>
                        </TouchableOpacity> : null}
            {
                props.sizeButton == 'compact' ?
                    props.typeButton == 'secondary' ?
                        <TouchableOpacity style={styles.mContainerSecondaryCompact} onPress={props.onPressButton}>
                            <Text style={styles.mTextSecondaryCompact}>
                                {props.text}
                            </Text>
                        </TouchableOpacity> :
                        <TouchableOpacity style={styles.mContainerPrimaryCompact} onPress={props.onPressButton}>
                            <Text style={styles.mTextPrimaryCompact}>
                                {props.text}
                            </Text>
                        </TouchableOpacity> : null}
            {
                props.sizeButton == 'no_line' ?
                    props.typeButton == 'secondary' ?
                        <TouchableOpacity style={styles.mContainerNoLine} onPress={props.onPressButton}>
                            <Text style={styles.mTextNoLine}>
                                {props.text}
                            </Text>
                        </TouchableOpacity> : null
                    : null}
        </>

    );
};
