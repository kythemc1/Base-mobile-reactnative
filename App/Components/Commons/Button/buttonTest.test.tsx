import renderer from 'react-test-renderer';
import {Button} from 'Components/Commons/Button/index';

test('testButton', () => {
    const component = renderer.create(<Button text={'test'} typeButton={'secondary'} sizeButton={'compact'}
        onPressButton={() => {
        }}/>).toJSON();
    expect(component).toMatchSnapshot();

}, 100);
