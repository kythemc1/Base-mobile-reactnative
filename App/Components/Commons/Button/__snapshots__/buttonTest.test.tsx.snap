// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testButton 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#143980",
      "borderRadius": 12,
      "borderWidth": 1,
      "height": 40.019999999999996,
      "justifyContent": "center",
      "opacity": 1,
      "width": 225,
    }
  }
>
  <Text
    style={
      {
        "color": "#143980",
        "fontSize": 12,
        "fontWeight": "400",
        "lineHeight": 16,
      }
    }
  >
    test
  </Text>
</View>
`;
